sources/web/datalab/polymer/modules/bigquery-file-manager/bigquery-file-manager.ts

```diff
@@ -92,19 +99,31 @@ class BigQueryFileManager implements FileManager {
     throw new UnsupportedMethod('getEditorUrl', this);
   }
 
+  private async _listMoreProjects(accumulatedProjects: ProjectResource[],
+      pageToken: string): Promise<DatalabFile[]> {
+    const response: HttpResponse<ListProjectsResponse> =
+        await GapiManager.bigquery.listProjects(pageToken);
+    const additionalProjects = response.result.projects || [];
+    const projects = accumulatedProjects.concat(additionalProjects);
+    if (response.result.nextPageToken) {
+      return this._listMoreProjects(projects, response.result.nextPageToken);
+    } else {
+      projects.sort((a: ProjectResource, b: ProjectResource) => {
```

Comment:
Curious how long this takes on a 1000+ projects?

Reply:
About 12 seconds for my 1200 projects. :-(

Reply:
Just the sort or the entire thing?